includes/MediaWiki/Api/EditEndpoint.php

```diff
@@ -137,27 +137,40 @@ public function run() {
 		$reconcileUrlProperty = new PropertyId( $inputReconcile['urlReconcile'] );
 		// For now this property must be of URL type
 		if ( $this->propertyDataTypeLookup->getDataTypeIdForProperty( $reconcileUrlProperty ) !== 'url' ) {
-			die( 'urlReconcile property must be of type url' );
+			throw new LocalizedHttpException(
+				MessageValue::new( 'wikibasereconcileedit-editendpoint-invalid-type-property-must-be-url' ),
+				400
+			);
 		}
 
 		// Get Item from input
 		if ( !array_key_exists( self::VERSION_KEY, $request->entity() ) ) {
-			die( 'entity input version must be specified in key ' . self::VERSION_KEY );
+			throw new LocalizedHttpException(
+				MessageValue::new( 'wikibasereconcileedit-editendpoint-unspecified-entity-input-version' ),
+				400
+			);
 		}
 		$inputEntityVersion = $request->entity()[self::VERSION_KEY];
 		if ( $inputEntityVersion === '0.0.1/full' ) {
 			$inputEntity = ( new FullWikibaseItemInput )->getItem( $request );
 		} elseif ( $inputEntityVersion === '0.0.1/minimal' ) {
 			$inputEntity = ( new MinimalItemInput( $this->propertyDataTypeLookup ) )->getItem( $request );
 		} else {
-			die( 'unknown entity input version' );
+			throw new LocalizedHttpException(
+				MessageValue::new( 'wikibasereconcileedit-editendpoint-invalid-entity-input-version' )
+					->textListParams( [ '0.0.1/full', '0.0.1/minimal' ] ),
```

Comment:
This parameter isn’t currently used in the message. If we want to include it, I think we should also include the `$inputEntityVersion`, and we’ll also need the number of allowed versions for pluralization (like in an earlier message). So the message text could be something like:

> Unknown entity input version $1; the only supported {{PLURAL:$3|version is|versions are}} $2.

Reply:
You are right. :sweat_smile: I didn't use the parameter. Now done. :)

Reply:
True, I didn't add it :sweat_smile: Done.